racket/src/expander/extract/simplify-defn.rkt

```diff
@@ -20,20 +21,90 @@
       (seteq)
       (set-union (car args) (apply union-all (cdr args)))))
 
-;; compute the variables that are the target of a set! in e
+; This function gets called when encountering a 'suspect letrec': some
+; letrec expression where the bound variable is used somewhere inside
+; the bound expression, and we can't proove (given the information
+; available here) that uses is safe: that the implicit mutation that
+; happens as the letrec defines the variables.
```

Comment:
This reads oddly to me, so I wonder if I missed something or if a clause is missing.

I expected to read `that use is safe: [clarify what "is safe means"]`, and I think I got `that use is safe: that the implicit mutation […].`, where I think I would have expected `is not observable` or something similar before the full stop.